web/src/unread_ops.ts

```diff
@@ -606,6 +608,14 @@ export function process_read_messages_event(message_ids: number[]): void {
         }
     }
 
+    if (
+        message_lists.current !== undefined &&
+        !message_lists.current.can_mark_messages_read() &&
```

Comment:
Do you have a reason to make it conditional on `!message_lists.current.can_mark_messages_read()`?